### PR TITLE
Multi peer: Sync with random peer

### LINKED ISSUE #3450: Implement ability to connect to a bitcoin node and determine if they support blockfilter services
https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki#node-operation

If we connect to a bitcoin node, but they don't support blockfilters over p2p, we shouldn't use that node for initial block download as they can't serve us block filters.

In the future, we may way want to connect to some nodes that don't serve block filters as they can still be used to retrieve full blocks.

### PR DESCRIPTION
**Pull request by solo1g**
Closes #3450.

### CODE REVIEW

File: node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala [hunk 1]
```diff
@@ -41,13 +42,24 @@ case class NeutrinoNode(
     copy(dataMessageHandler = dataMessageHandler)
   }
 
+  lazy val randomPeerMsgSenderWithCompactFilters: PeerMessageSender = {
```
Comment (benthecarman) [inline node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala]: this should be a `def` so we get a different peer every time

**Review by benthecarman — COMMENTED** [review]
This will also need to be done for selecting a peer in the `DataMessageHandler`. As this currently stands it will select a random peer to sync with to start and only use it for the duration. This can be done in a separate PR however

File: node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala (hunk 1 above)
Comment (solo1g) [reply node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala]: Ok, changing that. I was under the impression that we we would need to have same peer for getcfheaders and for syncing. Also, looking into data message handler, missed that one.

**Comment by Christewart** [comment]
Looks like there was a compile failure: https://github.com/bitcoin-s/bitcoin-s/pull/3454/checks?check_run_id=3189403551#step:5:142

**Comment by solo1g** [comment]
Yeah, this issue I have resolved and am running the tests locally now before pushing again. Would request review again once all is done.

**Comment by solo1g** [comment]
1 test fails on 2.12 which passes on 2.13. The failing one is unrelated to the changes here and I have confirmed that it passes locally, so it would probably pass if I run CI again but leaving that for now.

File: node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ import scala.concurrent.Future
 
 case class NeutrinoNode(
     nodePeer: Vector[Peer],
-    dataMessageHandler: DataMessageHandler,
+    var dataMessageHandler: DataMessageHandler,
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala]: What was the reason for changing this? This makes `dataMessageHandler` a publicly available mutable variable which is generally a bad idea.

File: node/src/main/scala/org/bitcoins/node/Node.scala [hunk 3]
```diff
@@ -45,6 +46,36 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   val peers: Vector[Peer]
 
+  val peerServices: mutable.Map[Peer, ServiceIdentifier] =
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/Node.scala]: This is a publicly available mutable variable. Can this be made `private`?

**Review by Christewart — CHANGES_REQUESTED** [review]

File: node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala (hunk 2 above)
Comment (solo1g) [reply node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala]: The reason for changing this here is for changing `updateDataMessageHandler` to what it is now. Earlier used `copy` to update and but did not copy the vals defined in `Node` trait that `NeutrinoNode` extends so this led to the `peerServices` map we had in `Node` trait to be initialized to empty. Also, there was one other place where states diverged in `DataMessageHandler` and led to errors in `handleDataPayload` of  `PeerMessageReceiver`. 

I get that this should not be public so made it private and added a getter for it.

File: node/src/main/scala/org/bitcoins/node/Node.scala (hunk 3 above)
Comment (solo1g) [reply node/src/main/scala/org/bitcoins/node/Node.scala]: Done

**Review by Christewart — APPROVED** [review]

**Pull request merged**